pandas/tseries/tdi.py

```diff
@@ -344,6 +344,17 @@ def _add_datelike(self, other):
             other = Timestamp(other)
             i8 = self.asi8
             result = i8 + other.value
+            # gh-14068: there is the possibility of addition overflow,
+            # which occurs when we add two very large positive numbers,
+            # resulting in a negative number.
+            #
+            # Since Timestamp objects can never have a negative value,
+            # we just need to check that the elements in result are also
+            # nonnegative when their corresponding elements in i8 are.
+            for x, y in zip(i8, result):
```

Comment:
Took a fresh look, and it seems like my question regarding masking is also relevant to this patch as well.  Should we be complaining about overflow in elements that are going to be masked off at the end anyways?